x/upgrade/client/cli/tx.go

```diff
@@ -0,0 +1,184 @@
+package cli
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/x/gov/client/cli"
+
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
+	"github.com/cosmos/cosmos-sdk/x/gov"
+	upgrade "github.com/cosmos/cosmos-sdk/x/upgrade/internal/types"
+)
+
+const (
+	// TimeFormat specifies ISO UTC format for submitting the upgrade-time for a new upgrade proposal
+	TimeFormat = "2006-01-02T15:04:05Z"
+
+	FlagUpgradeName   = "upgrade-name"
+	FlagUpgradeHeight = "upgrade-height"
+	FlagUpgradeTime   = "upgrade-time"
+	FlagUpgradeInfo   = "upgrade-info"
+)
+
+func parseArgsToContent(cmd *cobra.Command, cdc *codec.Codec) (gov.Content, error) {
+	title, err := cmd.Flags().GetString(cli.FlagTitle)
+	if err != nil {
+		return nil, err
+	}
+
+	description, err := cmd.Flags().GetString(cli.FlagDescription)
+	if err != nil {
+		return nil, err
+	}
+
+	name, err := cmd.Flags().GetString(FlagUpgradeName)
+	if err != nil {
+		return nil, err
+	}
+	if len(name) == 0 {
+		name = title
+	}
+
+	height, err := cmd.Flags().GetInt64(FlagUpgradeHeight)
+	if err != nil {
+		return nil, err
+	}
+
+	timeStr, err := cmd.Flags().GetString(FlagUpgradeTime)
+	if err != nil {
+		return nil, err
+	}
+
+	if height != 0 {
+		if len(timeStr) != 0 {
+			return nil, fmt.Errorf("only one of --upgrade-time or --upgrade-height should be specified")
+		}
+	}
+
+	var upgradeTime time.Time
+	if len(timeStr) != 0 {
+		upgradeTime, err = time.Parse(TimeFormat, timeStr)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	info, err := cmd.Flags().GetString(FlagUpgradeInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	plan := upgrade.Plan{Name: name, Time: upgradeTime, Height: height, Info: info}
+	content := upgrade.NewSoftwareUpgradeProposal(title, description, plan)
+	return content, nil
+}
+
+// GetCmdSubmitUpgradeProposal implements a command handler for submitting a software upgrade proposal transaction.
+func GetCmdSubmitUpgradeProposal(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "software-upgrade --upgrade-name [name] (--upgrade-height [height] | --upgrade-time [time]) (--upgrade-info [info]) [flags]",
```

Comment:
Transition `--upgrade-name` to an arg and this can be `software-upgrade [name] (--...`

Reply:
Fair enough, I made name an arg, and then removed the `upgrade-` prefix from the rest of the args. Take a look

Reply:
It still looks like a flag here...

I would prefer the CLI signature looked something like this (I am a bit opinionated about these things, sorry!):

```go
Use: "software-upgrade [upgrade-name] [height-or-rfc3339-time] [upgrade-info]
```

This would eliminate a number of flags and make the user experience much better. We have transitioned almost all of the `required` flag to args in the codebase to make for better command readability.